posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,572 @@
+        """
+        self.n_comp = n_comp
+        
+        # process training data 
+        self.initial = initial  # initial conditions in log space
+        self.rho_min = np.min(profiles,axis=1) 
+        rho_max = np.max(profiles,axis=1)
+        profiles_norm = (profiles-self.rho_min[:,np.newaxis])\
+                            /(rho_max-self.rho_min)[:,np.newaxis]  # minmax normalized profiles
+        self.pca = PCA(n_components=self.n_comp).fit(profiles_norm) # perform principal component analysis
+        weights_unscaled = self.pca.transform(profiles_norm)
+        self.scaling = np.std(weights_unscaled,axis=0)
+        self.weights = weights_unscaled/self.scaling  # scaled PCA weights
+        
+        # process testing data
+        self.valid_initial = valid_initial
+        self.valid_rho_min = np.min(valid_profiles,axis=1)
+        valid_rho_max = np.max(valid_profiles,axis=1)
+        valid_profiles_norm = (valid_profiles-self.valid_rho_min[:,np.newaxis])\
+                            /(valid_rho_max-self.valid_rho_min)[:,np.newaxis]
+        valid_weights_unscaled = self.pca.transform(valid_profiles_norm)
+        self.valid_weights = valid_weights_unscaled/self.scaling
+        
+        # instantiate models
+        self.model_prof = models.Sequential([
+            layers.Dense(15,input_dim=3,activation=None),
+            layers.Dense(15,input_dim=15,activation="relu"),
+            layers.Dense(15,input_dim=15,activation="relu"),
+            layers.Dense(15,input_dim=15,activation="tanh"),
+            layers.Dense(15,input_dim=15,activation="tanh"),
+            layers.Dense(10,input_dim=10,activation="tanh"),
+            layers.Dense(10,input_dim=10,activation="tanh"),
+            layers.Dense(n_comp,input_dim=10,activation=None)])
+                
+        self.model_rho = models.Sequential([
+            layers.Dense(10,input_dim=3,activation='relu'),
+            layers.Dense(10,input_dim=10,activation='relu'),
+            layers.Dense(10,input_dim=10,activation='tanh'),
+            layers.Dense(10,input_dim=10,activation='tanh'),
+            layers.Dense(10,input_dim=10,activation='tanh'),
+            layers.Dense(10,input_dim=10,activation='tanh'),
+            layers.Dense(1,activation=None)])
+        
+        self.model_IF = IFInterpolator()  # instantiate POSYDON initial-final interpolator object
+        self.model_IF.load(filename=IF_interpolator)
+                
+    def train(self,loss=losses.MeanSquaredError()):
+        """Trains NN models. 
+        Args: 
+            loss (object) : loss function for training.
+        """
+        print("training on PCA weights...")
+        
+        self.model_prof.compile(optimizers.Adam(clipnorm=1),loss=loss)
+        callback = tf.keras.callbacks.EarlyStopping(monitor="loss",patience=200)
+        history = self.model_prof.fit(self.initial,self.weights,
+                                      epochs=3000,callbacks=[callback],verbose=0,
```

Comment:
I would have more of these parameters be accessible for tweaking by the user. Depending on the data, it may take more than 3000 epochs to converge, and a different user may want to be more patient. I think that since this whole DL thing is new to most astro people, there will definitely be those curious in what's going on under the hood and some may want to play around with things, which is something I think we should encourage. This goes for all calls of the fit function.

Reply:
I made the parameters for the PCA weights and H mass fraction boundary points customizable. I don't think anyone could ever need more for the minimum density, since it's such a simple problem. usually only takes a few seconds to train.